js/post-select/containers/browse-filters.js

```diff
@@ -28,6 +28,7 @@ class PostBrowseFiltersContainer extends React.Component {
 			<PostBrowseFilters
 				formId={ this.state.id }
 				postTypeObjects={ this.props.postTypeObjects }
+				showDateFilters={ this.props.showDateFilters }
```

Comment:
Passing down this props all the way from the root component till the date component seems messy.
I wonder if we can use Context API here?
cc: @kadamwhite looking for your recommendation here too.

Reply:
@kadamwhite I've implemented the UI/UX changes you shared and also fixed the bug when switching dates. The outstanding thing from the feedback is this one which was shared by @ivankristianto. I'm not quite sure how to go about it yet.

Reply:
I like @ivankristianto's suggestion a lot, but since we're not using that currently, I feel that introducing Context here would feel out-of-scope for this ticket. We don't really want to have two competing paradigms at once: I'd rather proceed with the property passing method @Rayhatron is using, and then open a new ticket to refactor the components to use Context more heavily so that we can adjust things wholesale.

Reply:
Makes sense, I've created the new ticket for this.